### DISCUSSION TITLE
Central package version does not work for Uno.Mobile project with Visual Studio 2022 Preview

### DISCUSSION THREAD

**Discussion by inforithmics** [unoplatform/uno#8976] [Q&A]
### Current behavior

NU1008	Projects that use central package version management should not define the version on the PackageReference items but on the PackageVersion items: Uno.UI.RemoteControl;Xamarin.Google.Android.Material;Microsoft.VisualStudio.Threading.Analyzers;Microsoft.Net.Compilers.Toolset;IDisposableAnalyzers;SkiaSharp.Views.Uno;Uno.UI;Uno.UniversalImageLoader;Uno.UI.Adapter.Microsoft.Extensions.Logging;Microsoft.Extensions.Logging.Console;Microsoft.Extensions.Logging.	Mapsui.Samples.Uno.Mobile	C:\Users\Local Administrator\Documents\GitHub\Mapsui\Samples\Mapsui.Samples.Uno\Mapsui.Samples.Uno.Mobile\Mapsui.Samples.Uno.Mobile.csproj

### Expected behavior

No NU1008 Warning

### How to reproduce it (as minimally and precisely as possible)

1. Clone https://github.com/Mapsui/Mapsui
2. Run https://github.com/Mapsui/Mapsui/tree/master/Samples/Mapsui.Samples.Uno/Mapsui.Samples.Uno.Mobile Project in Visual Studio 2022 Preview
3. NU1008 warnings/errors appear, for maui no warnings/errors appear and it just works.

### Workaround

None found yet:
`<NoWarn>$(NoWarn);NU1008</NoWarn>`
Seems not to work.

Here I tried to fix the warnings/errors (And updated to the newest offical uno versions).
https://github.com/Mapsui/Mapsui/pull/1630

### Anything else we need to know?

Running on the github Actions with Visual Studio 2022 17.2 seems to work.
Here a sample run look for Mapsui.Samples.Unol.Mobile
https://github.com/Mapsui/Mapsui/runs/6715447452?check_suite_focus=true

**Comment by jeromelaban** [comment]
Thanks for the report! In your solution, the Uno projects are placed in a separate folder, which prevents the `Directory.Build.props` to work properly.

You'll need to add the following `Directory.Build.props` file to the `Samples/Mapsui.Samples.Uno` folder:
```xml
<Project ToolsVersion="15.0">

  <!-- https://docs.microsoft.com/en-us/visualstudio/msbuild/customize-your-build?view=vs-2019#use-case-multi-level-merging -->
  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(MSBuildThisFileDirectory)../'))" />

</Project>
```

The documentation mentions [something similar](https://docs.microsoft.com/en-us/nuget/consume-packages/central-package-management#central-package-management-rules).

**Reply by thomasstockermc** [reply]
Thanks a lot this helped. I didn't know that deeper folder structures cause problems for Directory.Build.props.

**Reply by jeromelaban** [reply]
Happy to help!
